client/src/stores/workspaceStore.ts

```diff
@@ -0,0 +1,82 @@
+import { create } from "zustand";
+import {
+  deleteWorkSpace,
+  getUserWorkspaces,
+  postAddNewWorkspace,
+  putEditWorkSpaceName,
+} from "../api/workspacesApi";
+import { AxiosResponse } from "axios";
+
+export interface workSpaceType {
+  name: string;
+  id: string;
+  createdAt?: string;
+  updatedAt?: string;
+}
+
+interface workspaceStore {
+  workspaces: workSpaceType[];
+  openWorkspace: workSpaceType;
+  setWorkspaces: (workspaces: workSpaceType[]) => void;
+  getWorkspaces: () => Promise<AxiosResponse<workSpaceType[], any>>;
+  changeBetweenWorkSpaces: (workSpaceId: string) => void;
+  editWorkSpaceName: (name: string, id: string) => Promise<void>;
+  createWorkspace: (name: string) => void;
+  deleteWorkspace: (id: string) => Promise<void>;
+}
+
+export const useWorkspaceStore = create<workspaceStore>((set, get) => ({
+  workspaces: [],
+  openWorkspace: {
+    name: "",
+    id: "",
+  },
+
+  getWorkspaces: () => {
+    return getUserWorkspaces();
```

Comment:
Why don't you set the workspaces once you call this function? it would reduce code in other functions in this store.